File: ServiceApp/src/main/java/com/example/ServiceApp/dto/DocumentDataDto.java
package com.example.ServiceApp.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;


@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DocumentDataDto {

    private Long id;
    private Long customerId;
    private String customerName;
    private String cui;
    private LocalDate contractDate;
    private Integer monthOfWarranty;
    private Integer monthOfWarrantyHandPieces;
    private String numberOfContract;

    @Builder.Default
    private List<DocumentEquipmentDto> equipments = new ArrayList<>();

    @Builder.Default
    private List<DocumentProductDto> products = new ArrayList<>();

    @Builder.Default
    private List<DocumentTrainedPersonDto> trainedPersons = new ArrayList<>();

    private LocalDate signatureDate;
    private String contactPerson;
}

File: service-interface/src/app/dtos/documentDataDto.ts
export interface DocumentProductDto {
  id?: number;
  productId?: number;
  productName?: string;
  productCod?: string;
  quantity?: number;
  sortOrder?: number;
}

export interface DocumentEquipmentDto {
  id?: number;
  equipmentId?: number;
  equipmentName?: string;
  productCode?: string;
  serialNumber?: string;
  sortOrder?: number;
}

export interface DocumentTrainedPersonDto {
  id?: number;
  trainedPersonName?: string;
  jobFunction?: string;
  phone?: string;
  email?: string;
  signatureBase64?: string;
  sortOrder?: number;
}

export class DocumentDataDto {
  id?: number;
  customerName?: string;
  cui?: string;
  monthOfWarranty?: number;
  monthOfWarrantyHandPieces?: number;
  numberOfContract?: string;
  contactPerson?: string;
  customerId?: number;
  contractDate?: string;
  signatureDate?: string;
  signatureBase64?: string;
  equipments?: DocumentEquipmentDto[];
  products?: DocumentProductDto[];
  trainedPersons?: DocumentTrainedPersonDto[];
}
